File: src/Avatar.Package/Avatar.StaticFactory.cs
namespace Avatars
{
    partial class Avatar
    {
        static Avatar()
        {
            AvatarFactory.Default = new StaticAvatarFactory();
            OnInitialized();
        }

        /// <summary>
        /// Invoked after the default <see cref="AvatarFactory.Default"/> 
        /// is initialized.
        /// </summary>
        static partial void OnInitialized();
    }
}

File: src/Avatar.Package/Avatar.StaticFactory.vb
Namespace Global.Avatars
    Partial Friend Class Avatar
        Shared Sub New()
            AvatarFactory.[Default] = New StaticAvatarFactory
            OnInitialized()
        End Sub

        ''' <summary>
        ''' Invoked after the default <see cref="AvatarFactory.Default"/> 
        ''' is initialized.
        ''' </summary>
        Partial Private Shared Sub OnInitialized()

        End Sub
    End Class
End Namespace